modules/io/io/oss_io_adaptor.h

```diff
@@ -93,13 +95,11 @@ class OSSIOAdaptor : public IIOAdaptor {
   }
 
  private:
-  Status listAllObjects(const std::string& prefix, const std::string& suffix);
+  Status getTotalSize(size_t& size);
 
-  void selectObjects();
+  Status readLine(std::string& line, size_t& cursor);
 
-  Status producerRoutine();
-
-  Status getObjectToBuffer(const std::string& object_name, std::string& block);
+  Status getRange(const size_t begin, const size_t end, std::string& content);
```

Comment:
Is there other option that we could void using `std::string` to wrap the result? to avoid coping as best as we can.